--- obistools-0.0.10.orig/R/check_onland.R	2021-02-08 13:19:45.000000000 -0300
+++ obistools-0.0.10/R/check_onland.R	2023-03-08 14:47:58.342251496 -0300
@@ -50,7 +50,7 @@
     i <- which(!is.na(over(sp, land)))
   } else {
     shoredistances <- lookup_xy(data, shoredistance = TRUE, grids = FALSE, areas = FALSE, asdataframe = TRUE)
-    i <- which(as.vector(shoredistances) < (-1*buffer))
+    i <- which(as.vector(unlist(shoredistances)) < (-1*buffer))
   }
   if (report) {
     if (length(i) > 0) {
